lib/scan-includes.js

```diff
@@ -0,0 +1,110 @@
+const realFetchUrl = require("./fetch-url");
+const GITHUB_SERVICE = require("./services").GITHUB;
+
+
+/** Scans for recursively included paths from a root specification.
+ *
+ * @param {string} repositoryUrl The base URL of the repository. The function
+ * won't fetch anything outside of this scope. This must end with a `/`.
+ * @param {string} rootSpec A path relative to the `repositoryUrl` pointing to
+ * the root specification in the `specType` format.
+ * @param {"bikeshed"|"respec"} specType
+ *
+ * @returns {Promise<Set<string>>} Paths relative to `repositoryUrl` that are
+ * recursively included by the specification.
+ */
+module.exports = async function scanIncludes(repositoryUrl, rootUrl, specType, fetchUrl = realFetchUrl) {
+    if (!repositoryUrl.endsWith('/')) {
+        throw new Error(`${repositoryUrl} must end with a '/'.`);
+    }
+    let includeRE;
+    switch (specType) {
+        case "bikeshed":
+            includeRE = /^path:([^\n]+)$/gm;
+            break;
+        case "respec":
+            includeRE = /data-include="([^"]+)"|data-include='([^']+)'/g;
```

Comment:
I think we can safely assume authors haven't used mixed quotes, or include pathnames containing quotes.
We'll have a single capture group and we won't need to loop below.

```suggestion
            includeRE = /data-include=["']([^"']+)["']/g;
```
![image](https://user-images.githubusercontent.com/8426945/81971101-2e785c00-963e-11ea-90db-b9d42689d988.png)

Or:
```suggestion
            includeRE = /data-include=["']?([^"'>]+)["']?/g;
```
![image](https://user-images.githubusercontent.com/8426945/81971287-7b5c3280-963e-11ea-9556-53a9b6812094.png)